### PR TITLE
Performance improvements

### LINKED ISSUE #275: Startup time of dojo CLI is high
**Enhancement**

At the moment the startup time for running `dojo` is around 2000+ milliseconds:

screenshot_20190204_224842

It looks like the majority of the time is spent loading modules.

Here's the profile as a ZIP:
[CPU-20190204T224612.cpuprofile.zip](https://github.com/dojo/cli/files/2830087/CPU-20190204T224612.cpuprofile.zip)

Package Version: 5.0.1-pre

**Code**

`dojo`

**Expected behavior:**

The CLI loads snappily

**Actual behavior:**

There is a noticeable delay

### PR DESCRIPTION
**Pull request by JamesMilnerUK**
**Type:** bug / feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [x] Unit or Functional tests are included in the PR

**Description:**

This PR aims to do two things:

* Only glob and load commands that match the command that is passed in (rather than loading all)
* When using `dojo`, `dojo --help`, `dojo -h` do not actually load the commands in question as they will never be used 

Supersedes #282 #281 

Resolves #275 

Before:

Screenshot_20190520_171817

After:
Screenshot_20190520_171724

### CODE REVIEW

File: tests/unit/loadCommands.ts [hunk 1]
```diff
@@ -162,6 +163,24 @@ registerSuite('loadCommands', {
...
 				}
 			}
+		},
+		'specific command': {
+			afterEach() {
+				mockModule.destroy();
+				testSandbox.restore();
+			},
+			tests: {
+				async 'should not load commands that arent specified group'() {
```
Comment (maier49) [inline tests/unit/loadCommands.ts]: "arent in"?

File: tests/unit/loadCommands.ts [hunk 2]
```diff
@@ -162,6 +163,24 @@ registerSuite('loadCommands', {
...
+					process.argv = ['node', 'dojo.js', 'bar'];
+					const installedPaths = await enumInstalledCommands(goodConfig);
+					assert.isTrue(installedPaths.length === 0);
+				},
+				async 'should not load commands that are in specified group'() {
```
Comment (maier49) [inline tests/unit/loadCommands.ts]: "should load"?

File: src/loadCommands.ts [hunk 3]
```diff
@@ -15,8 +15,10 @@ export function isEjected(groupName: string, command: string): boolean {
  * @returns {Promise<string []>} the paths of all installed commands
  */
 export async function enumerateInstalledCommands(config: CliConfig): Promise<string[]> {
+	const [, , command] = process.argv;
```
Comment (maier49) [inline src/loadCommands.ts]: Will `command` be defined here if I just run `dojo build`?

Comment (JamesMilnerUK) [reply src/loadCommands.ts]: It will indeed as it's `['node', 'path/to/script/', 'build']

File: tests/unit/loadCommands.ts (hunk 2 above)
Comment (JamesMilnerUK) [reply tests/unit/loadCommands.ts]: Good spot

File: src/loadCommands.ts (hunk 3 above)
Comment (matt-gadd) [reply src/loadCommands.ts]: this is my fault I think. the variable should be named `group` as it narrows by group not command.

Comment (JamesMilnerUK) [reply src/loadCommands.ts]: I'll update

**Pull request merged**